app/codes/statemanager.py

```diff
@@ -0,0 +1,238 @@
+import json
+import datetime
+import time
+import hashlib
+
+
+def is_wallet_valid(cur, address):
+    wallet_cursor = cur.execute('SELECT wallet_public FROM wallets WHERE wallet_address=?', (address, ))
+    wallet = wallet_cursor.fetchone()
+    if wallet is None:
+        return False
+    return True
+
+
+def transfer_tokens_and_update_balances(cur, sender, reciever, tokencode, amount):
+    sender_balance = get_wallet_token_balance(cur, sender, tokencode)
+    print("Sender is ", sender, " and their balance is ", sender_balance)
+    reciever_balance = get_wallet_token_balance(cur, reciever, tokencode)
+    print("Receiver is ", reciever, " and their balance is ", reciever_balance)
+    sender_balance = sender_balance - amount
+    reciever_balance = reciever_balance + amount
+    print("Amount is ", amount)
+    print("Updating sender's balance with ", sender_balance)
+    print("Updating reciever's balance with ", reciever_balance)
+    update_wallet_token_balance(cur, sender, tokencode, sender_balance)
+    update_wallet_token_balance(cur, reciever, tokencode, reciever_balance)
+    sender_balance = get_wallet_token_balance(cur, sender, tokencode)
+    print("Sender is ", sender, " and their balance is ", sender_balance)
+    reciever_balance = get_wallet_token_balance(cur, reciever, tokencode)
+    print("Receiver is ", reciever, " and their balance is ", reciever_balance)
+
+
+def update_wallet_token_balance(cur, wallet_address, token_code, balance):
+    cur.execute(f'''INSERT OR REPLACE INTO balances
+				(wallet_address, tokencode, balance)
+				 VALUES (?, ?, ?)''', (wallet_address, token_code, balance))
+
+
+def update_trust_score(cur, personid1, personid2, new_score, tstamp):
```

Comment:
i presume this is unique even though timestamp changes? on account of the primary key (person1-person2) being repeated?